fitbenchmarking/fitbenchmarking
# Added the precommit autoupdate workflow using the GitHub Actions bot

## Description of work

Fixes #1390 




## Testing Instructions


I tested this on a private repo to make sure that the PR was being created correctly. I did however had to make sure that the GitHub-actions[bot] had write permissions on the repo. There wouldn't be an easy way to test it expect merging and running the workflow and making sure that the branch protection rules allow for PRs to be created.

<img width="791" alt="image" src="https://github.com/user-attachments/assets/9d5f711b-8634-4fda-8742-6d51dbdef93c">

## Checklist
